bff/pkg/server/server.go

```diff
@@ -833,3 +844,141 @@ func isValidClusterPrefix(clusterPrefix string) bool {
 	}
 	return false
 }
+
+type Snapshot struct {
+	S3Location  string    `json:"s3_location"`
+	CollectedAt time.Time `json:"collected_at"`
+}
+
+func snapshots_handler(dataPath string) http.HandlerFunc {
+	snapshotTableNames := map[string]string{
+		"full":    "snapshots",
+		"compact": "compact_snapshots",
+	}
+
+	openDatabase := func(dataPath string) (*sql.DB, error) {
+		dbPath := filepath.Join(dataPath, "crystaldb-collector.db")
+		db, err := sql.Open("sqlite3", dbPath+"?mode=ro")
+		if err != nil {
+			return nil, fmt.Errorf("failed to open database: %v", err)
+		}
+		return db, nil
+	}
+
+	return func(w http.ResponseWriter, r *http.Request) {
+		limitStr := r.URL.Query().Get("limit")
+		snapshotType := r.URL.Query().Get("type")
+
+		limit := 1 // default to 1 if not specified
+		if limitStr != "" {
+			parsedLimit, err := strconv.Atoi(limitStr)
+			if err != nil || parsedLimit < 1 {
+				http.Error(w, "Invalid limit parameter", http.StatusBadRequest)
+				return
+			}
+			limit = parsedLimit
+		}
+
+		// Validate snapshot type
+		tableName, ok := snapshotTableNames[snapshotType]
+		if !ok {
+			http.Error(w, "Invalid type parameter. Must be 'compact' or 'full'", http.StatusBadRequest)
+			return
+		}
+
+		// Open database connection
+		db, err := openDatabase(dataPath)
+		if err != nil {
+			http.Error(w, "Failed to open database", http.StatusInternalServerError)
+			return
+		}
+		defer db.Close()
+
+		// Query the snapshots from SQLite
+		query := `
+			SELECT s3_location, collected_at
+			FROM (
+				SELECT s3_location, collected_at
+				FROM ` + tableName + `
+				ORDER BY collected_at DESC
+				LIMIT ?
+			)
+			ORDER BY collected_at ASC`
```

Comment:
There's no need for the nested query.

```suggestion
			SELECT s3_location, collected_at
			FROM ` + tableName + `
			ORDER BY collected_at DESC
			LIMIT ?`
```

Reply:
This change would reverse the order of the results.

Reply:
OK, I thought limit is always 1, but I see 1 is just a default limit. SG. Please ignore my comment.